src/tokenizedSubaccounts/TSAPreDeposit.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.18;
+
+import "openzeppelin/token/ERC20/extensions/IERC20Metadata.sol";
+import {IERC20Metadata} from "openzeppelin/token/ERC20/extensions/IERC20Metadata.sol";
+import {Ownable2Step} from "openzeppelin/access/Ownable2Step.sol";
+import {IERC20Metadata} from "openzeppelin/token/ERC20/extensions/IERC20Metadata.sol";
+
+contract TSAPreDeposit is Ownable2Step {
+  IERC20Metadata public depositAsset;
+  mapping(address => uint) public deposits;
```

Comment:
one ask, can you create a read function for passing in a `List[address]` and getting back `List[balances]`?

I can use this to static call RPC and get all deposit balances in one go. 

Something like this in the actual strategy contract would be super helpful too.